cmd/manager/main.go

```diff
@@ -200,12 +230,21 @@ func main() {
 
 	unpacker := &source.ContainersImageRegistry{
 		BaseCachePath: filepath.Join(cachePath, "unpack"),
-		SourceContext: &types.SystemContext{
-			DockerCertPath: caCertDir,
-			OCICertPath:    caCertDir,
-			AuthFilePath:   authFilePathIfPresent(setupLog),
-		},
-	}
+		SourceContextFunc: func(logger logr.Logger) (*types.SystemContext, error) {
+			srcContext := &types.SystemContext{
+				DockerCertPath: caCertDir,
+				OCICertPath:    caCertDir,
+			}
+			if _, err := os.Stat(authFilePath); err == nil {
+				logger.Info("using available authentication information for pulling image")
+				srcContext.AuthFilePath = authFilePath
+			} else if os.IsNotExist(err) {
+				logger.Info("no authentication information found for pulling image, proceeding without auth")
```

Comment:
Just had an interesting thought. This behavior means that an enterprising person could mount a file at `authFilePath`, NOT set `--global-pull-secret` and have this work.

If we want to avoid unintended usage like this, we may want to:
1. Make this code conditional on the `--global-pull-secret` flag being set, AND
2. Doing something like this to get a random file name every time we start:
   ```
   authFilePath := filepath.Join(os.TempDir(), fmt.Sprintf("%s-%s.json", authFilePrefix, apimachineryrand.String(8))
   ```

Reply:
Slightly related: I was actually thinking about us using an optional secret mount instead of a flag: https://kubernetes.io/docs/concepts/configuration/secret/#restriction-secret-must-exist

But I think this will limit users to having to create a global pull secret in the same namespace as operator-controller.

Reply:
>But I think this will limit users to having to create a global pull secret in the same namespace as operator-controller.

Yea that's not a viable option for us. 

Also incorporated Joe's suggestions.

Reply:
Can we mark this thread resolved?